apps/entropy-tester/src/index.ts

```diff
@@ -0,0 +1,194 @@
+import fs from "node:fs/promises";
+
+import type { PrivateKey } from "@pythnetwork/contract-manager/core/base";
+import { toPrivateKey } from "@pythnetwork/contract-manager/core/base";
+import { EvmEntropyContract } from "@pythnetwork/contract-manager/core/contracts/evm";
+import { DefaultStore } from "@pythnetwork/contract-manager/node/store";
+import type { Logger } from "pino";
+import { pino } from "pino";
+import type { ArgumentsCamelCase, InferredOptionTypes } from "yargs";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+type LoadedConfig = {
+  contract: EvmEntropyContract;
+  interval: number;
+};
+
+function timeToSeconds(timeStr: string): number {
+  const match = /^(\d+)([hms])$/i.exec(timeStr);
+  if (!match?.[1] || !match[2])
+    throw new Error("Invalid format. Use formats like '6h', '15m', or '30s'.");
+
+  const value = Number.parseInt(match[1], 10);
+  const unit = match[2].toLowerCase();
+
+  switch (unit) {
+    case "h": {
+      return value * 3600;
+    }
+    case "m": {
+      return value * 60;
+    }
+    case "s": {
+      return value;
+    }
+    default: {
+      throw new Error("Unsupported time unit.");
+    }
+  }
+}
+
+type ParsedConfig = {
+  "chain-id": string;
+  interval: string;
+};
+
+async function loadConfig(configPath: string): Promise<LoadedConfig[]> {
+  const configs = (await import(configPath, {
+    with: { type: "json" },
+  })) as { default: ParsedConfig[] };
+  const loadedConfigs = configs.default.map((config) => {
+    const interval = timeToSeconds(config.interval);
+    const contracts = Object.values(DefaultStore.entropy_contracts).filter(
+      (contract) => contract.chain.getId() == config["chain-id"],
+    );
+    if (contracts.length === 0 || !contracts[0]) {
+      throw new Error(
+        `Can not find the contract for chain ${config["chain-id"]}, check contract manager store.`,
+      );
+    }
+    if (contracts.length > 1) {
+      throw new Error(
+        `Multiple contracts found for chain ${config["chain-id"]}, check contract manager store.`,
+      );
+    }
+    return { contract: contracts[0], interval };
+  });
+  return loadedConfigs;
+}
+
+async function testLatency(
+  contract: EvmEntropyContract,
+  privateKey: PrivateKey,
+  logger: Logger,
+) {
+  const provider = await contract.getDefaultProvider();
+  const userRandomNumber = contract.generateUserRandomNumber();
+  const requestResponse = (await contract.requestRandomness(
+    userRandomNumber,
+    provider,
+    privateKey,
+    true, // with callback
+  )) as {
+    transactionHash: string;
+    events: {
+      RequestedWithCallback: {
+        returnValues: {
+          sequenceNumber: string;
+        };
+      };
+    };
+  };
```

Comment:
Same comment -- don't typecast, instead parse with zod.  Typecasting is unsafe and it breaks all the type safety the type checker attempts to provide.